torch/nested/__init__.py

```diff
@@ -106,6 +106,9 @@ def as_nested_tensor(
             return torch._nested_tensor_from_tensor_list(ts, dtype, None, device, None)
     elif layout == torch.jagged:
         if isinstance(ts, Tensor):
+            if device is None:
+                device = ts.device
```

Comment:
We don't need this fallback for dtype?

Reply:
good Q and the reason why we don't is because `offsets` is constructed with type `torch.int64` regardless of the input `dtype`